lib/cubits/purchase/purchase_cubit.dart

```diff
@@ -16,25 +16,23 @@ class PurchaseCubit extends Cubit<PurchaseState> {
   PurchaseCubit({required this.paymentHandler, required this.product})
       : super(const PurchaseInitial());
 
-  Future<void> payWithMobilePay() async {
+  Future<void> pay() async {
+    sl<FirebaseAnalyticsEventLogging>().beginCheckoutEvent(product);
+
     if (state is PurchaseInitial) {
-      sl<FirebaseAnalyticsEventLogging>().beginCheckoutEvent(product);
       emit(const PurchaseStarted());
 
       final either = await paymentHandler.initPurchase(product.id);
 
-      either.fold(
-        (error) => emit(PurchaseError(error.reason)),
-        (payment) async {
-          if (payment.status != PaymentStatus.error) {
-            emit(PurchaseProcessing(payment));
-            await paymentHandler
-                .invokePaymentMethod(Uri.parse(payment.deeplink));
-          } else {
-            emit(PurchasePaymentRejected(payment));
-          }
-        },
-      );
+      either.fold((error) => emit(PurchaseError(error.reason)), (payment) {
+        if (payment.status == PaymentStatus.completed) {
+          emit(PurchaseCompleted(payment));
+        } else if (payment.status == PaymentStatus.awaitingPayment) {
+          emit(PurchaseProcessing(payment));
+        } else {
+          emit(PurchasePaymentRejected(payment));
+        }
+      });
```

Comment:
move each case to its own line:


either.fold((error) => emit(PurchaseError(error.reason)), (payment) {
        if (payment.status == PaymentStatus.completed) {
          emit(PurchaseCompleted(payment));
        } else if (payment.status == PaymentStatus.awaitingPayment) {
          emit(PurchaseProcessing(payment));
        } else {
          emit(PurchasePaymentRejected(payment));
        }
      },
  );

Reply:
Current formatting follows the autoformatter. Is it the extra comma you are going for, otherwise I don't see what you wnat here?

Reply:
yes, the extra command puts each case to its own line